src/java/org/apache/cassandra/index/sai/StorageAttachedIndex.java

```diff
@@ -493,8 +493,7 @@ public long getEstimatedResultRows()
     @Override
     public boolean isQueryable(Status status)
     {
-        // consider unknown status as queryable, because gossip may not be up-to-date for newly joining nodes.
-        return status == Status.BUILD_SUCCEEDED || status == Status.UNKNOWN;
+        return status == Status.BUILD_SUCCEEDED;
```

Comment:
The main reason I changed this is that it made my more lightweight mechanism for avoiding the middle node in the test matrix work with less effort. However, I'm not entirely sure why we've ever considered UNKNOWN to be a queryable state. If gossip hasn't propagated yet, was the idea to just fire queries at the new node and hope the index is queryable locally? Is doing the simpler thing (only using BUILD_SUCCEEDED) going to cause a major availability problem?

CC @mike-tr-adamson @adelapena

Reply:
I think the consensus from some discussion on ASF Slack is that we can still attempt queries against UNKNOWN replicas, but only if there aren't enough BUILD_SUCCEEDED replicas present.

CC @jasonstack @pkolaczk @JeremiahDJordan